### PR TITLE
feat(connector): [Dlocal] Use connector_request_reference_id as reference to the connector

### PR DESCRIPTION
**Pull request by nikita812**
…ctor #2297

## Type of Change

- [x ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by prasunna09 — COMMENTED** [review]
Hi @nikita812, Please follow [commit message guidelines ](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#commit-message-guidelines)

**Comment by nikita812** [comment]
Thanks for the commands, I am working on it.

File: crates/router/src/connector/dlocal/transformers.rs [hunk 1]
```diff
@@ -185,7 +185,7 @@ impl TryFrom<&types::PaymentsCaptureRouterData> for DlocalPaymentsCaptureRequest
             authorization_id: item.request.connector_transaction_id.clone(),
             amount: item.request.amount_to_capture,
             currency: item.request.currency.to_string(),
-            order_id: item.payment_id.clone(),
+            connector_request_reference_id: item.connector_request_reference_id.clone(),
```
Comment (prasunna09) [inline crates/router/src/connector/dlocal/transformers.rs:188]:
```suggestion
            order_id: item.connector_request_reference_id.clone(),
```

File: crates/router/src/connector/dlocal/transformers.rs [hunk 2]
```diff
@@ -392,7 +392,7 @@ impl<F, T>
 #[derive(Default, Debug, Serialize)]
 pub struct RefundRequest {
     pub amount: String,
-    pub payment_id: String,
+    pub connector_request_reference_id: String,
```
Comment (prasunna09) [inline crates/router/src/connector/dlocal/transformers.rs:395]: same here, please revert this change

File: crates/router/src/connector/dlocal/transformers.rs [hunk 3]
```diff
@@ -403,7 +403,7 @@ impl<F> TryFrom<&types::RefundsRouterData<F>> for RefundRequest {
         let amount_to_refund = item.request.refund_amount.to_string();
         Ok(Self {
             amount: amount_to_refund,
-            payment_id: item.request.connector_transaction_id.clone(),
+            connector_request_reference_id: item.request.connector_request_reference_id.clone(),
```
Comment (prasunna09) [inline crates/router/src/connector/dlocal/transformers.rs:406]: here acc to [dev docs](https://docs.dlocal.com/reference/make-a-refund) of Dlocal we don't need this change, we have to send connector_transaction_id as payment id to make refund for the respective payment.
Hence please revert back this change

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Comment by SargamPuram** [comment]
hey @prasunna09 , I recently raised a PR that was just merged today which solves something like this #2446 . I guess this PR #2491 is of a duplicate issue please have a look.  As @srujanchikke had already closed the PR #2490 , that @nikita812 had raised previously since the issue was originally assigned to me. I request you to look into this please.

**Comment by prasunna09** [comment]
> hey @prasunna09 , I recently raised a PR that was just merged today which solves something like this #2446 . I guess this PR #2491 is of a duplicate issue please have a look. As @srujanchikke had already closed the PR #2490 , that @nikita812 had raised previously since the issue was originally assigned to me. I request you to look into this please.

Hi @SargamPuram, the issue you resolved is related to connector response reference id but this is connector request reference id, both are different.

**Comment by SargamPuram** [comment]
> > hey @prasunna09 , I recently raised a PR that was just merged today which solves something like this #2446 . I guess this PR #2491 is of a duplicate issue please have a look. As @srujanchikke had already closed the PR #2490 , that @nikita812 had raised previously since the issue was originally assigned to me. I request you to look into this please.
> 
> Hi @SargamPuram, the issue you resolved is related to connector response reference id but this is connector request reference id, both are different.

Hi @prasunna09 I apologise for the confusion. Thanks for the clarification.

**Comment by swangi-kumari** [comment]
Hello @nikita812 ,
Let us know if you are facing any issue, happy to help:)

**Comment by nikita812** [comment]
Hello to all maintainers
Due to breakdown of my machine I wasn't able to continue with this issue and it might take a bit more time to get my machine fixed. I thank all of you for being considerate to what issues I was facing, its rare to see that but sorry for the inconvenience caused.

**Pull request closed without merging**